pkg/services/sqlstore/migrations/playlist_mig.go

```diff
@@ -44,3 +34,32 @@ func addPlaylistMigrations(mg *Migrator) {
 		{Name: "title", Type: DB_Text, Nullable: false},
 	}))
 }
+
+func addPlaylistUIDMigration(mg *Migrator) {
+	// Replacing auto-incremented playlistIDs with string UIDs
+	mg.AddMigration("Add UID column to playlist", NewAddColumnMigration(playlistV2(), &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 80, Nullable: false, Default: "0",
+	}))
```

Comment:
This, along with the unique index created below, would cause an issue for backwards compatibility. If we need to roll back the application version, the application would not be able to write more than 1 playlist to the DB without hitting the uniqueness constraint (the default `"0"` value would be used multiple times).

[There are functions like `UNIQUE INDEX` in MySQL for example that would help with this](https://www.mysqltutorial.org/mysql-unique/). I would suggest considering some alternative here.

Reply:
that's a very good point but I honestly don't know which is scarier: seize the unique constrain in the common case or breaking going backwards. FYI that's similar for the datasources:
https://github.com/grafana/grafana/blob/0a23299878056f88d61527a88bcce9788379cb5b/pkg/services/sqlstore/migrations/datasource_mig.go#L139

https://github.com/grafana/grafana/blob/0a23299878056f88d61527a88bcce9788379cb5b/pkg/services/sqlstore/migrations/datasource_mig.go#L152

Only compromise I can think of, would be to enforce this in the backend instead of the DB level. We will discuss it with the team and I'll be back.